Add meeting notes from September (#86)
python/docs-community: docs/monthly-meeting/2023-09.md

# Documentation Community Team Meeting (September 2023)

- **Date:** 2023-09-05
- **Time:** [19:00 UTC](https://arewemeetingyet.com/UTC/2023-09-05/19:00/Docs%20Meeting)
- **This HackMD:** https://hackmd.io/@encukou/pydocswg1
- [**Discourse thread**](https://discuss.python.org/t/32355) (for September)
- [**Meeting reports**](https://docs-community.readthedocs.io/en/latest/monthly-meeting/index.html) (the latest one might be an [**unmerged PR**](https://github.com/python/docs-community/pulls))
- **Calendar event:** (send your e-mail to Mariatta for an invitation)
- **How to participate:**
  -  Go to [Google Meet](https://meet.google.com/dii-qrzf-wkw) and ask to be let in.
  -  To edit notes, click the “pencil” or “split view” button on the [HackMD document](https://hackmd.io/@encukou/pydocswg1). You need to log in (e.g. with a GitHub account).

By participating in this meeting, you are agreeing to abide by and uphold the [PSF Code of Conduct](https://www.python.org/psf/codeofconduct/).
Please take a second to read through it!

## Roll call

(Name / `@GitHubUsername` / Discord if different)
- CAM Gerlach / `@CAM-Gerlach`

> - Daniele / `@EvilDMP`

Reviewer: ```suggestion
- Daniele Procida / `@EvilDMP`
```

- Petr Viktorin / `@encukou`
- Adam Turner / `@AA-Turner`
- Hugo van Kemenade / `@hugovk`
- Maciej Olko / `@m-aciek` / `@m_aciek`

> - Ryan / `@ryan-duve`

Reviewer: ```suggestion
- Ryan Duve / `@ryan-duve`
```

- Marcus Sherman / `@betteridiot`

> - Erlend / `@erlend-aasland`

Reviewer: ```suggestion
- Erlend Aasland / `@erlend-aasland`
```

- Ege Akman / `@egeakman`
- Ezio Melotti / `@ezio-melotti`

> ## Introductions
> 
> - Maciej, from Poland, works on Polish translation
> 

Reviewer: We didn't capture these well, I'd remove the section
```suggestion
```

Author: We mostly have our introductions in previous meeting notes, Maciej was new so I think we can keep it for context. @m-aciek what do you think?

Reviewer 2: I don't have an opinion, I'd defer to you

Author: Keeping the introductions

## Reports and celebrations

> - [Hugo] [docs.python.org](https://docs.python.org) now has a dark theme! 🌒

Reviewer: ```suggestion
- [Hugo] [docs.python.org](https://docs.python.org/3/) now has a dark theme! 🌒
```

- [Adam] New PEP helper & validator ([`check-peps.py`](https://github.com/python/peps/commit/814ceede975056cc56a019bbed62b29a6a875830))

- [Adam] [Datamodel headings](https://github.com/python/cpython/pull/108146) for readability

- [Adam] Work to [remove references to Distutils](https://github.com/python/cpython/issues/92584) in the documentation ahead of the 3.12.0 release

## Discussion

- [Hugo] python-docs-theme:
  - Need a release of 2023.8 to fix some dark theme bugs: [python/python-docs-theme#152](https://github.com/python/python-docs-theme/issues/152)
  - Have a PR ready to release using Trusted Publishers, need someone to set it up on PyPI: [python/python-docs-theme#148](https://github.com/python/python-docs-theme/pull/148)
  - These are both waiting on people who can make releases. Adam: should we review the group of people who can make releases, and look to expand it? Potentially ties in to a [Python organisation](https://github.com/python/steering-council/issues/207) on PyPI.

- [Ege/Hugo/CAM] Updates and discussion on Plausible hosting:
  - PSF folks opposed to self-hosting
  - Paid hosting is a *little* expensive

>   - We got a 15% discount from them, which would be \$1400-\$1500 yearly

Reviewer: Are these public numbers? (If not, too late, but perhaps we should avoid putting in a public website; a GH PR is at least a little less discoverable)

Author: Oof, should we redact the numbers? Even if we do redact the numbers it'll still be visible in the commit history.

Author: Agreed to keep it

  - People brought up several motivating use cases already for this, including:
    - Translation prioritization for languages/pages
    - Automated translations
    - Determine how many people use backported branches
    - Determining if old pages are still needed/used
    - Potentially determine common broken links, maybe even fragment ids?
    - Prioritisation for documentation

>   - [Previous Core Development budget thread](https://discuss.python.org/t/23507)

Reviewer: This seems out of place, could we give it some context?

Author: IIRC, the idea was "we have the budget, why not use it".

  - Next steps: CAM to write to Deb from the PSF with a summary of this discussion

> - [Maciej] Translations platform – trying to initiate discussion ([Discourse thread](https://discuss.python.org/t/docs-translation-platform/29940))

Reviewer: We should capture the outcome of this chat, which was to get in touch with the PSF (with Cam's support)

  - Transifex has cut off shared memory (between projects) for open source tier, Rafael from Brazilian Portugese team implemented a workaround
  - Weblate – based on Django, open license, used by PyPI
  - The POC of Python docs translations/Weblate integration is functional, I'm working on improving it
  - Python docs don't fit into free open source tier on Weblate – the initial proposition is 1500€/year (50% discount)
  - There is an option to self-host a Weblate instance
  - (Both platforms support using machine learning translation tools)

- [Ege] New bootstrapper for starting new Python docs translations
  - Issues with the [current tool](https://github.com/JulienPalard/python-docs-cookiecutter):
    - Unmaintained / inactive
    - Uses third-party modules

>   - Add a link from the [DevGuide](https://devguide.python.org/documentation/translating/#starting-a-new-translation) to this [repository](https://github.com/egeakman/python-docs-bootstrapper)

Author: I don't think there is an *official* way of writing or capitalizing it. All variations (devguide, DevGuide, devguide) have been used before in this repository.

- [devguide](https://github.com/python/docs-community/blob/main/docs/monthly-meeting/2023-01.md?plain=1#L74)
- [Devguide](https://github.com/python/docs-community/blob/main/docs/monthly-meeting/2023-01.md?plain=1#L73)
- [DevGuide](https://github.com/python/docs-community/blob/main/docs/monthly-meeting/2022-11.md?plain=1#L35)

Though I'm not against choosing one of them.

>   - Add a link from the [DevGuide](https://devguide.python.org/documentation/translating/#starting-a-new-translation) to this [repository](https://github.com/egeakman/python-docs-bootstrapper)

Reviewer 3: There's something like [90 of devguide](https://github.com/search?q=org%3Apython+%2F%28%3F-i%29%5B%5E%5C%2F%5Ddevguide%2F&type=code), [17 of Devguide](https://github.com/search?q=org%3Apython+%2F%28%3F-i%29Devguide%2F&type=code) and [three of DevGuide](https://github.com/search?q=org%3Apython+%2F%28%3F-i%29DevGuide%2F&type=code) in the Python org. 

In the guide's repo, it something like [18](https://github.com/search?q=repo%3Apython%2Fdevguide++%2F%28%3F-i%29%5B%5E%5C%2F%5Ddevguide%2F&type=code), [two](https://github.com/search?q=repo%3Apython%2Fdevguide+%2F%28%3F-i%29Devguide%2F&type=code) and [zero](https://github.com/search?q=repo%3Apython%2Fdevguide+%2F%28%3F-i%29DevGuide%2F&type=code). 

Let's use devguide.

>   - Add a link from the [DevGuide](https://devguide.python.org/documentation/translating/#starting-a-new-translation) to this [repository](https://github.com/egeakman/python-docs-bootstrapper)

Reviewer 3: And the https://github.com/python/devguide README says:

> This guide covers how to contribute to CPython. It is known by the nickname of "the devguide" by the Python core developers.

>   - Add a link from the [DevGuide](https://devguide.python.org/documentation/translating/#starting-a-new-translation) to this [repository](https://github.com/egeakman/python-docs-bootstrapper)

Author: Sounds reasonable to me, thanks!

>   - Add a link from the [devguide](https://devguide.python.org/documentation/translating/#starting-a-new-translation) to this [repository](https://github.com/egeakman/python-docs-bootstrapper)

Reviewer: Maybe preface with 'action' or 'next steps'

>   - [PEP 545](https://peps.python.org/pep-0545/#repository-for-po-files) mentions the per-language repositories for ``.po`` files

Author: I don't think this is relevant, should we remove it?

Author: Let's keep this here for the record.

>   - [PEP 545](https://peps.python.org/pep-0545/#repository-for-po-files) mentions the per-language repositories for `.po` files

Reviewer: I think I wrote this, the discussion briefly turned to having all translations managed online through a provider and I was looking for the citation that we should at least save the .mo files to Git. But probably irrelevant for these notes

Author: Removed

- [CAM] Quick update on reStructured Data
  - Summary
    - Mechanism to extract, process and output structured, machine-readible data from the Python docs
    - Initial focus on deprecations/removals per user requests, then additions/changes and finally other annotations (supported platforms, stable ABI, audit events, etc.)
    - Output a JSON per version and language for tool and API consumption

>     - Also output a table of all deprecations/removals (or other changes) in the 

Author: I believe this got incomplete at the last minute. @CAM-Gerlach can you take a look?

Reviewer 4: Oops, might have left that incomplete, sorry. Added [a suggestion](https://github.com/python/docs-community/pull/86/files#r1335058819) (couldn't add it here as this was considered outdated) to fix this, thanks.

>     - Also output a table of all deprecations/removals (or other changes) in the

Reviewer 4: Seems I might have left this incomplete, oops...

```suggestion
    - Also output tables of all/specific deprecations/removals (or other changes) in the What's New and on a new dedicated Deprecations/Removals page.
```

Author: Thanks!

  - Progress
    - Further considerable work on design document, schema, etc.
    - Developed initial working implementation and prototyped/tested most planned features

>     - Discussed current approach at some length with `@AA-Turner` and currently making some significant changes based on his feedback

Reviewer: Having a common name is inconvenient when choosing a username...

```suggestion
    - Discussed current approach at some length with Adam and currently making some significant changes based on his feedback
```

    - Next steps: Re-scoping to focus on the basic features needed
  - Learn more
    - [HackMD detailed design document and schema](https://hackmd.io/jF8awEd3TuKh1Ia5Hqa_8g?view#TransformsProcessors)
    - [Initial working prototype implementation](https://github.com/python/cpython/compare/main...CAM-Gerlach:cpython:add-structured-deprecation-support)

- [Hugo] Python 3.12.0 is out on 2023-10-02! Any finishing touches needed for [What’s New In Python 3.12](https://docs.python.org/3.12/whatsnew/3.12.html)?
  - [CAM] Was planning to finish this at the core dev sprints, but didn't realize the final release date was so much earlier than last year
  - [CAM] Wanted to be able to use the new reStructured Data directive to list deprecations, removals, additions and changes in the appropriate places following the existing format, but too late for that now
  - [CAM] May be able to use it to help guide the process, but timeline is going to be tight

- [Daniele] How much appetite/consensus is there for an iterative workflow? [Discussion in Discord](https://discord.com/channels/935215565872693329/935215566334079058/1138743377962225794)

>   - Daniele can adapt the Discord messages into a Discourse thread and post that for further feedback and consensus

Reviewer: I think @encukou suggested he may have the time to do this, I can't remember if it was decided though

Reviewer 5: It's in my TODO pile, but I wouldn't mind if Daniele gor to it first :)

Author: Added you both
